const categoriaNegocio = require('./negocio/categoria_negocio')

async function main() {

    // INSERIR //
    // try {
    //     const categoriaInserida1 = await categoriaNegocio.inserir({nome: "cat2"})
    //     console.log("Categoria inserida: ", categoriaInserida1)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }
    
    // try {
    //     const categoriaInserida2 = await categoriaNegocio.inserir()
    //     console.log("Categoria inserida: ", categoriaInserida2)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }

    // LISTAR //
    // const listaCategoria = await categoriaNegocio.listar()
    // console.log("Lista categorias: ", listaCategoria)

    // BUSCAR POR ID //
    // try {
    //     const buscarId1 = await categoriaNegocio.buscarPorId(1)
    //     console.log("Buscar por ID: ", buscarId1)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }
    
    // try {
    //     const buscarId2 = await categoriaNegocio.buscarPorId(5)
    //     console.log("Buscar por ID: ", buscarId2)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }

    // ATUALIZAR //
    // try {
    //     const atualizarCategoria1 = await categoriaNegocio.atualizar(2, {nome: 'cat22'})
    //     console.log("Atualizar: ", atualizarCategoria1)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }

    // try {
    //     const atualizarCategoria2 = await categoriaNegocio.atualizar(1)
    //     console.log("Atualizar: ", atualizarCategoria2)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }

    // try {
    //     const atualizarCategoria2 = await categoriaNegocio.atualizar(5, {nome: 'cat4'})
    //     console.log("Atualizar: ", atualizarCategoria2)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }


    // DELETAR //
    // try {
    //     const deletarCategoria1 = await categoriaNegocio.deletar(2)
    //     console.log("Deletar: ", deletarCategoria1)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }

    // try {
    //     const deletarCategoria2 = await categoriaNegocio.deletar(20)
    //     console.log("Deletar: ", deletarCategoria2)
    // } catch (error) {
    //     console.log("Erro: ", error)
    // }
    
    
}

main()